gobblin-modules/gobblin-crypto/src/main/java/org/apache/gobblin/crypto/GPGFileDecryptor.java

```diff
@@ -113,23 +117,35 @@ public InputStream decryptFile(InputStream inputStream, InputStream keyIn, Strin
       throw new IllegalArgumentException("secret key for message not found.");
     }
 
+    ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+
     try (InputStream clear = pbe.getDataStream(
         new JcePublicKeyDataDecryptorFactoryBuilder().setProvider(BouncyCastleProvider.PROVIDER_NAME).build(sKey))) {
 
       JcaPGPObjectFactory pgpFact = new JcaPGPObjectFactory(clear);
       Object pgpfObject = pgpFact.nextObject();
 
-      if (pgpfObject instanceof PGPCompressedData) {
-        PGPCompressedData cData = (PGPCompressedData) pgpfObject;
-        pgpFact = new JcaPGPObjectFactory(cData.getDataStream());
+      while (pgpfObject != null) {
+        if (pgpfObject instanceof PGPCompressedData) {
+          PGPCompressedData cData = (PGPCompressedData) pgpfObject;
+          pgpFact = new JcaPGPObjectFactory(cData.getDataStream());
+          pgpfObject = pgpFact.nextObject();
+        }
+
+        if (pgpfObject instanceof PGPLiteralData) {
+          Streams.pipeAll(((PGPLiteralData) pgpfObject).getInputStream(), outputStream);
+        } else if (pgpfObject instanceof PGPOnePassSignatureList) {
+          throw new PGPException("encrypted message contains PGPOnePassSignatureList message - not literal data.");
+        } else if (pgpfObject instanceof PGPSignatureList) {
+          throw new PGPException("encrypted message contains PGPSignatureList message - not literal data.");
+        } else {
+          throw new PGPException("message is not a simple encrypted file - type unknown.");
+        }
         pgpfObject = pgpFact.nextObject();
-        PGPLiteralData ld = (PGPLiteralData) pgpfObject;
-        return ld.getInputStream();
-      } else if (pgpfObject instanceof PGPOnePassSignatureList) {
-        throw new PGPException("encrypted message contains a signed message - not literal data.");
-      } else {
-        throw new PGPException("message is not a simple encrypted file - type unknown.");
       }
+      return new ByteArrayInputStream(outputStream.toByteArray());
```

Comment:
This won't work well for large files since it requires materializing the whole stream to create another one. Can you  instead wrap the logic above into a new inputstream type so that the inputstream can be incrementally decrypted into the output stream as data is requested?

Reply:
I guess we can use piped streams: https://stackoverflow.com/questions/1225909/most-efficient-way-to-create-inputstream-from-outputstream

Reply:
This bug is blocking the customer, so the current fix will be allowed in. I filed the following ticket to track this issue. GOBBLIN-293
Remove stream materialization in GPGFileDecryptor